openhcl/underhill_attestation/src/crypto.rs

```diff
@@ -129,9 +131,24 @@ pub fn pkcs11_rsa_aes_key_unwrap(
     unwrapping_rsa_key: &Rsa<Private>,
     wrapped_key_blob: &[u8],
 ) -> Result<Rsa<Private>, Pkcs11RsaAesKeyUnwrapError> {
-    let modulus_size = unwrapping_rsa_key.size();
-    let wrapped_aes_key = &wrapped_key_blob[..modulus_size as usize];
-    let wrapped_rsa_key = &wrapped_key_blob[modulus_size as usize..];
+    let modulus_size = unwrapping_rsa_key.size() as usize;
+
+    let (wrapped_aes_key, wrapped_rsa_key) = wrapped_key_blob
+        .split_at_checked(modulus_size)
+        .ok_or_else(|| {
+            Pkcs11RsaAesKeyUnwrapError::UndersizedWrappedKeyBlob(format!(
```

Comment:
Suggest creating an unique error for each case instead of defining customized string

Reply:
Attempted to address in https://github.com/microsoft/openvmm/pull/587/commits/2ee6763e2385338fc712d6f1387340133f43f97c